src/libostree/ostree-repo-pull.c

```diff
@@ -502,8 +539,13 @@ idle_worker (gpointer user_data)
   scan_one_metadata_object (pull_data, checksum, scan_data->objtype,
                             scan_data->path, scan_data->recursion_depth,
                             scan_data->requested_ref, pull_data->cancellable, &error);
-  check_outstanding_requests_handle_error (pull_data, &error);
-  scan_object_queue_data_free (scan_data);
+
+  if (should_retry_request (error, scan_data->n_retries_remaining--))
```

Comment:
Why do we need to check for network errors here? `scan_one_metadata_object` either enqueues a request or just scans a local object, right?

Reply:
Hmm, indeed. I blame over-enthusiastic grepping for calls to `check_outstanding_requests_handle_error()`.